src/concurrency/vector_clock.rs

```diff
@@ -40,9 +45,40 @@ impl From<u32> for VectorIdx {
 /// clock vectors larger than this will be stored on the heap
 const SMALL_VECTOR: usize = 4;
 
-/// The type of the time-stamps recorded in the data-race detector
-/// set to a type of unsigned integer
-pub type VTimestamp = u32;
+/// The time-stamps recorded in the data-race detector consist of both
+/// a 32-bit unsigned integer which is the actual timestamp, and a `Span`
+/// so that diagnostics can report what code was responsible for an operation.
+#[derive(Clone, Copy, Debug, Eq)]
```

Comment:
The `Eq` is suspicious. This will also compare spans. Is that really what we want? Why is it even needed?

EDIT: Oh, PartialEq contains all the logic. yeah that is confusing...

Reply:
I could have sworn clippy wanted the Eq derived. I agree it's simpler explicitly.